vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/test/java/com/vaadin/flow/component/sidenav/tests/SideNavItemTest.java

```diff
@@ -388,6 +383,126 @@ public void expandAndCollapse_isCollapsed() {
+        sideNavItem.setQueryParameters(queryParameters);
+
+        Assert.assertEquals("path?" + queryParameters.getQueryString(),
+                sideNavItem.getPath());
+    }
+
+    @Test
+    public void setPath_setQueryParameters_setQueryParametersNull_parametersRemovedFromPath() {
+        sideNavItem.setPath("path");
+        sideNavItem.setQueryParameters(
+                new QueryParameters(Map.of("k1", List.of("v11", "v12"))));
+
+        sideNavItem.setQueryParameters(null);
+
+        Assert.assertEquals("path", sideNavItem.getPath());
+    }
+
+    @Test
+    public void createFromComponent_setQueryParameters_pathContainsParameters() {
+        try (MockedStatic<ComponentUtil> mockComponentUtil = Mockito
+                .mockStatic(ComponentUtil.class)) {
+            mockComponentUtil.when(() -> ComponentUtil.getRouter(Mockito.any()))
+                    .thenReturn(router);
+
+            SideNavItem item = new SideNavItem("test", TestRoute.class);
+
+            QueryParameters queryParameters = new QueryParameters(Map.of("k1",
+                    List.of("v11", "v12"), "k2", List.of("v21", "v22")));
+            item.setQueryParameters(queryParameters);
+
+            Assert.assertEquals("foo/bar?" + queryParameters.getQueryString(),
+                    item.getPath());
+        }
+    }
+
+    @Test
+    public void createFromComponent_setQueryParameters_updateQueryParameters_pathIsUpdated() {
+        try (MockedStatic<ComponentUtil> mockComponentUtil = Mockito
+                .mockStatic(ComponentUtil.class)) {
+            mockComponentUtil.when(() -> ComponentUtil.getRouter(Mockito.any()))
+                    .thenReturn(router);
+
+            SideNavItem item = new SideNavItem("test", TestRoute.class);
+            item.setQueryParameters(
+                    new QueryParameters(Map.of("k1", List.of("v11", "v12"))));
+
+            QueryParameters queryParameters = new QueryParameters(
+                    Map.of("k2", List.of("v21", "v22")));
+            item.setQueryParameters(queryParameters);
+
+            Assert.assertEquals("foo/bar?" + queryParameters.getQueryString(),
+                    item.getPath());
+        }
+    }
+
+    @Test
+    public void createFromComponent_setQueryParameters_setQueryParametersNull_parametersRemovedFromPath() {
+        try (MockedStatic<ComponentUtil> mockComponentUtil = Mockito
+                .mockStatic(ComponentUtil.class)) {
+            mockComponentUtil.when(() -> ComponentUtil.getRouter(Mockito.any()))
+                    .thenReturn(router);
+
+            SideNavItem item = new SideNavItem("test", TestRoute.class);
+            item.setQueryParameters(
+                    new QueryParameters(Map.of("k1", List.of("v11", "v12"))));
+
+            item.setQueryParameters(null);
+
+            Assert.assertEquals("foo/bar", item.getPath());
+        }
+    }
+
+    @Test
+    public void createFromComponentAndRouteParameters_pathContainsParameters() {
```

Comment:
It may be good to also have an explicit test for the `setPath(Class<? extends Component> view, RouterParameters parameters)` to ensure the existence of this method overload.

Reply:
Added a test.